File: Binary Search/codeforces_1928B.cpp

// Problem link ===>> https://codeforces.com/problemset/problem/1928/B
// submission link ===>> https://codeforces.com/contest/1928/submission/249340016

#include<bits/stdc++.h>
using namespace std;

void asquare(){
    long long n;
    cin >> n;

    vector<long long> arr(n);
    for (int i = 0; i < n; i++){
        cin >> arr[i];
    }

    set<long long> uniqueValues(arr.begin(), arr.end());
    vector<long long> sortedValues(uniqueValues.begin(), uniqueValues.end());

    long long ans = 0;
    for (int i = 0; i < sortedValues.size(); i++)
    {
        long long x = sortedValues[i];
        auto itt = lower_bound(sortedValues.begin(), sortedValues.end(), x + n);
        long long distance = itt - sortedValues.begin() - i;
        ans = max(ans, distance);
    }

    cout << ans << endl;
}

int main(){
    int t;
    cin >> t;

    while(t--){
        asquare();
    }

    return 0;
}

File: Binary Search/codeforces_1928B.py

# Problem link ===>> https://codeforces.com/problemset/problem/1928/B

from collections import Counter
from typing import List, Union


def asquare() -> None:
    n = int(input())
    arr: List[int] = list(map(int, input().split()))

    value_counts: Counter[Union[int, None]] = Counter(arr)
    unique_values = sorted(value_counts.keys())

    maximum_distance = 0
    for x in unique_values:
        try:
            distance = value_counts[x + n]
        except KeyError:
            distance = 0
        maximum_distance = max(maximum_distance, distance)

    print(maximum_distance)


if __name__ == "__main__":
    t = int(input())
    for _ in range(t):
        asquare()
